apps/emqx_resource/src/emqx_resource.erl

```diff
@@ -204,10 +203,7 @@ query(InstId, Request) ->
 %% it is the duty of the Module to apply the `after_query()` functions.
 -spec query(instance_id(), Request :: term(), after_query()) -> Result :: term().
 query(InstId, Request, AfterQuery) ->
-    case get_instance(InstId) of
-        {ok, _Group, #{status := connecting}} ->
-            query_error(connecting, <<"cannot serve query when the resource "
-                "instance is still connecting">>);
+    case emqx_resource_manager:ets_lookup(InstId) of
         {ok, _Group, #{status := disconnected}} ->
```

Comment:
Q: if the ETS table only holds `connected` instances, perhaps this case is now impossible?